[AzureMonitorExporter] Propagate node attributes

pcwiese opened the pull request:
These are attributes that apply to any Span and so should be propagated.
Specifically set the ai.cloud.name and ai.cloud.instance tags on the
envelope to provide a better Azure Monitor Application Map experience.

pjanotti reviewed (approved):
LGTM

SergeyKanzhelev commented on exporter/azuremonitorexporter/traceexporter.go:
```diff
@@ -653,6 +658,39 @@ func (exporter *traceExporter) pushTraceData(
+				// Assumes the node level attribute key/values correspond to:
+				// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md
+				if serviceName, ok := traceData.Node.Attributes[resourceAttributeKeyServiceName]; ok {
+					envelope.Tags[contracts.CloudRole] = serviceName
```
perhaps also prepend `service.namespace` to is.

pcwiese replied:
good idea.

tigrannajaryan commented on exporter/azuremonitorexporter/traceexporter.go:
```diff
@@ -71,6 +71,12 @@ const (
+	// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md
+	resourceAttributeKeyServiceName       = "service.name"
+	resourceAttributeKeyServiceNamespace  = "service.namespace"
+	resourceAttributeKeyServiceInstanceID = "service.instance.id"
```
These are already defined in https://github.com/open-telemetry/opentelemetry-collector/blob/master/translator/conventions/opentelemetry.go
Any reason for redefining and not importing?

pcwiese replied:
No, I'll definitely import.

pcwiese commented:
any other comments here?

tigrannajaryan commented on exporter/azuremonitorexporter/go.mod:
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/golang/groupcache v0.0.0-20191027212112-611e8accdfc9 // indirect
 	github.com/golang/protobuf v1.3.2
-	github.com/open-telemetry/opentelemetry-collector v0.2.1-0.20191016224815-dfabfb0c1d1e
+	github.com/open-telemetry/opentelemetry-collector v0.2.1-0.20191217213608-3cac55bf869f
```
Please rebase and revert this change. Collector is now at 0.2.2.

pcwiese committed:
Propagate node attributes

These are attributes that apply to any Span and so should be propagated.
Specifically set the ai.cloud.name and ai.cloud.instance tags on the
envelope to provide a better Azure Monitor Application Map experience.
